### ISSUE TITLE
Question: Why 'grunt serve' binds to localhost host ONLY?

### ISSUE THREAD

**Issue by verrol** [DaftMonk/generator-angular-fullstack#1114]
I am new to Angular-fullstack, but I have used a number of other generators to get here. Why is it, that with AFS, I can't do 'grunt serve' and then load the app from another computer in my network? 

When I run nmap from another host to see which ports are opened on the computer where my app is running. Port 9000 doesn't show up, even though the browser on the computer can certainly use it. 

I then modified the development env to use my computer IP address, the one I get from the router. It builds, port 9000 is visible from outside, but it doesn't work. 

Neither too does 'grunt serve:dist' work even on the localhost. I get a number of 404 error as the browser tries to fetch the page from localhost.

I guess I am missing something simple. But I never had any of these problems with the other generators, but I really would like to use AFS instead of MEAN.js.

I would appreciate any pointer on this issue or steps to fix.  Oh, I did some googling and saw others have had the same problem.

Thanks.

.v

PS: If this was done intentionally, why? Since it prevents users from testing on multiple devices during development.

**Comment by Awk34** [comment]
There is nothing from the generator preventing you from connecting from a different computer on your LAN.

**Comment by kingcody** [comment]
@verrol are you using the latest version of the generator (`2.1.1`)? If not perhaps #1080 might solve your issue.

**Comment by verrol** [comment]
@kingcody Thanks for the pointer. I have the latest and I checked config/environment/index.js. I have that change uses 'localhost'. It still doesn't work for me. Only removing the host/ip works.

May be it is my network or computer environment. I tried this simple example and I could reproduce the problem, with and without host/ip. It certainly seems for me anyway, that .listen() doesn't work with host/ip.

Example code:
server.js:
---------------------
var app = require('express')();
var server = require('http').Server(app);

var port = 9001;

server.listen(port, function(){
  console.log('Express server listening on %d port', port);
});

app.get('/', function (req, res) {
   res.sendFile(__dirname + '/index.html');
});
